### DISCUSSION TITLE
egs_brachy make compiling error in windows

### DISCUSSION THREAD

**Discussion by ROUZII3** [nrc-cnrc/EGSnrc#1103] [General]
I am new to EGS and seeking assistance with installing EGS_brachy on my Windows 10 system to conduct simulations in brachytherapy. While following the EGS_brachy manual, I encountered an error during the configuration of EGSnrc using the configuration shell script (./HEN_House/scripts/configure). Despite following the provided instructions and attempting to adjust the GCC version, the compilation of egs_c_utils.c and egs_c_utils.h failed. And it gave me this recommandations:
Creating egs_c_utils.h and egs_c_utils.c ... OK
Compiling egs_c_utils.c ... Failed
[the.error.msg.txt](https://github.com/nrc-cnrc/EGSnrc/files/14500886/the.error.msg.txt)
[configure.log](https://github.com/nrc-cnrc/EGSnrc/files/14500887/configure.log)
Following the invaluable advice and instructions provided by @rtownson, I proceeded to configure EGSnrc using the GUI installer for Windows. Upon successful installation, I meticulously followed the prescribed steps outlined in the EGS_brachy documentation for existing EGSnrc installations. However, upon attempting to compile the software using the make command in Git Bash, I encountered an error.
As this software is crucial for conducting simulations in brachytherapy, I am earnestly seeking assistance in resolving this issue to ensure the seamless installation and functionality of the EGS_brachy environment. Any guidance or support in this matter would be greatly appreciated.
[egs_brachy make error.txt](https://github.com/nrc-cnrc/EGSnrc/files/14501158/egs_brachy.make.error.txt)

**Comment by mchamberland** [comment]
Good news / bad news:

1. [This is a known issue](https://github.com/clrp-code/egs_brachy/issues/59).
2. I don't know how to resolve this issue properly and I do not have reliable access to Windows to sort this out. If you're comfortable with Git, your best bet is to roll back to a previous commit, for example, [this one.](https://github.com/clrp-code/egs_brachy/commit/ebfbe7f911710f737e1540a8e9ba46622317c988)

I cannot stress enough that if you have access to Linux over Windows, you should absolutely be using Linux. It will make your life much easier in terms of installing and maintaining EGSnrc.

If you have access to a Mac (especially an Apple Silicon Mac), then that would also be a good option and it will be easier for me to help.

**Reply by ROUZII3** [reply]
I appreciate your prompt response and the valuable advice provided. However,Is there no recourse available to address this problem, or are there any suggestions or recommendations for seeking assistance?

**Reply by mchamberland** [reply]
Yes, if you're familiar with Git, you should roll back to the commit I linked to.

If you are not familiar with Git, you will need to wait a bit until I have time to give you instructions. But using the commit I linked to should resolve the issue for you. Hopefully. 🤞

**Reply by ROUZII3** [reply]
i will try my best, thank you

**Comment by mchamberland** [comment]
By the way, @rtownson if you have any input on this DLLImport error on Windows, please share your wisdom with me. I tried creating an egs_brachy-specific uniform run control object to allow outputting gzip files, but that DLLImport shows up on Windows and I have no idea where to even start.

**Reply by rtownson** [reply]
I think you don't need `EGS_EXPORT` in your class. That's for creating a shared library object, which doesn't seem necessary if you're just using the class locally for egs_brachy.

This line:
```
class EGS_EXPORT EB_UniformRunControl : public EGS_RunControl {
```

**Reply by mchamberland** [reply]
I thought I had tried removing that EGS_EXPORT, but now I'm not sure. I'll try it again. Thank you!
